### PR TITLE
(IR interpreter) Add more intrinsics

### PR DESCRIPTION
**Pull request by protonu**

### CODE REVIEW

File: torch/csrc/jit/tensorexpr/ir.h [hunk 1]
```diff
@@ -787,6 +795,16 @@ class Intrinsics : public CallNode<Intrinsics> {
...
         return "remainder";
+      case kLgamma:
+        return "lgamma";
+      case kExpm1:
+        return "expm1";
+      case kErfc:
+        return "erfc";
+      case kFrac:
+        return "frac";
+      case kReciprocal:
```
Comment (resistor) [inline torch/csrc/jit/tensorexpr/ir.h]: Should we have this, or just use division to represent it?

**Comment by resistor** [comment]
Looks fine otherwise

File: torch/csrc/jit/tensorexpr/ir.h (hunk 1 above)
Comment (protonu) [reply torch/csrc/jit/tensorexpr/ir.h]: I'll use division. I'll update this PR

**Review by bertmaher — APPROVED** [review]

File: torch/csrc/jit/passes/tensorexpr_fuser.cpp [hunk 2]
```diff
@@ -588,6 +598,31 @@ class TensorExprKernel {
             });
       } break;
 
+      case aten::sigmoid: {
```
Comment (protonu) [inline torch/csrc/jit/passes/tensorexpr_fuser.cpp]: @bertmaher , @resistor Can you guys please take a quick look at Relu, Neg, Reciprocal and Sigmoid, and let me know if it looks okay

File: torch/csrc/jit/passes/tensorexpr_fuser.cpp [hunk 3]
```diff
@@ -588,6 +598,31 @@ class TensorExprKernel {
...
+        return Compute("aten_sigmoid", texprDims(v), [](const Expr& a) {
+          return Expr(1.0f) / exp(a);
```
Comment (bertmaher) [inline torch/csrc/jit/passes/tensorexpr_fuser.cpp]: is this actually the right expression for sigmoid?  Isn't it `1 / (1 + exp(-a))`?  But if it's not the right expression, how are the tests passing?

Comment (protonu) [reply torch/csrc/jit/passes/tensorexpr_fuser.cpp]: Thanks for catching that. That's not the right expression. Let me check the tests.

Comment (protonu) [reply torch/csrc/jit/passes/tensorexpr_fuser.cpp]: The tests were not working.
It's because torch.rand(1024, dtype=float), creates float64 (double) tensors and were not taking our code path. Modifying the tests throws a lot of interesting mismatches between the eager mode and our implementations. I am debugging this.

File: torch/csrc/jit/passes/tensorexpr_fuser.cpp (hunk 2 above)
Comment (protonu) [reply torch/csrc/jit/passes/tensorexpr_fuser.cpp]: @bertmaher @resistor  can you guys take another look at this reimplementation. For Sigmoid small fp differences show up, for which I had to change rtol and atol values in numpy.allclose()

File: test/test_tensorexpr.py [hunk 4]
```diff
@@ -441,20 +505,35 @@ def test_abs(x, y):
...
+        test_lgamma,        
+        test_sigmoid,
+        test_reciprocal,
+        test_neg,
+        test_relu,
     }
-    rand_a = torch.rand(1024, dtype=float)
-    rand_b = torch.rand(1024, dtype=float)
-    zeros = torch.zeros(1024, dtype=float)
+    rand_a = torch.rand(1024, dtype=torch.float)
```
Comment (protonu) [inline test/test_tensorexpr.py:620]: @bertmaher @resistor  I had to change type to torch.float to make sure doubles were not being used. This caused fp discrepencies in many of the instrinsic functions for which I had to relax rtol and atol for numpy.allclose. Let me know what you guys think in terms of bitwise inaccuracy

**Review by bertmaher — APPROVED** [review]
Looks good to me.  Can you add a note to our TODO doc that we should compare our numerics to eager mode and to GraphFuser?  IMO we should try to be within the tolerances of other implementations within pytorch, but shouldn't necessarily shoot for something "better" (since that might not be possible, short of writing our own math lib).

**Comment by protonu** [comment]
With all the code refactoring, this PR got too messy. I'll close it for a new one. I'll create a quip for the FP issues.

**Review by Krovatkin — COMMENTED** [review]
@protonu a quick q, did you intend to update fbgemm?
